go/vt/vttablet/tabletserver/vstreamer/vstreamer.go

```diff
@@ -1046,6 +1053,75 @@ func (vs *vstreamer) extractRowAndFilter(plan *streamerPlan, data []byte, dataCo
 		}
 		pos += l
 
+		// If the column is a CHAR based type with a binary collation (e.g. utf8mb4_bin) then the
+		// actual column type is included in the second byte of the event metadata while the
+		// event's type for the field is BINARY. This is true for ENUM and SET types.
+		var mysqlType uint16
+		if sqltypes.IsQuoted(plan.Table.Fields[colNum].Type) {
+			mysqlType = plan.TableMap.Metadata[colNum] >> 8
+		}
+		// Convert the integer values in the binlog event for any SET and ENUM fields into their
+		// string representations.
+		if plan.Table.Fields[colNum].Type == querypb.Type_ENUM || mysqlType == mysqlbinlog.TypeEnum {
+			// Add the mapping JiT in case we haven't properly received and processed a table map
+			// event to initialize it.
+			if plan.EnumSetValuesMap == nil {
+				if err := addEnumAndSetMappingstoPlan(plan.Plan, plan.Table.Fields, plan.TableMap.Metadata); err != nil {
+					return false, nil, false, fmt.Errorf("failed to build ENUM and SET column integer to string mappings: %v", err)
+				}
+			}
+			// ENUM columns are stored as an unsigned 16-bit integer as they can contain a maximum
+			// of 65,535 elements (https://dev.mysql.com/doc/refman/en/enum.html) with the 0 element
+			// reserved for any integer value that has no string mapping.
+			iv, err := value.ToUint16()
```

Comment:
Do we cover the case where `value` stands for `NULL`?

Reply:
Good point! I'll add that now.

Reply:
This has been added.